ckanext/unaids/blueprints/ape_data_receiver.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from flask import Blueprint, redirect, request, jsonify
+import requests
+from ckan.common import _, g
+from ckan.lib.helpers import url_for
+from ckan.plugins import toolkit
+
+from urllib.parse import urlparse, urlencode
+
+
+
+ape_data_receiver = Blueprint("ape_data_receiver", __name__)
+
+@ape_data_receiver.route('/ape_data_receiver', methods=['GET'])
+def receive():
+    # import pydevd_pycharm
+    # pydevd_pycharm.settrace('172.17.0.1', port=9999, stdoutToServer=True, stderrToServer=True)
+    if not g.user:
+        return toolkit.abort(403, _('You must be logged in to access this page'))
+    else:
+        parsed_url = urlparse(toolkit.config.get('ckanext.saml2auth.idp_metadata.remote_url'))
+        base_url = parsed_url.scheme + "://" + parsed_url.netloc + "/authorize"
+        client_id = toolkit.config.get('ape_client_id')
+        client_secret = toolkit.config.get('ape_client_secret')
+        redirect_url = toolkit.config.get('ape_callback_url')
+        state = toolkit.config.get('ape_state')
+        audience = url_for('ape_data_receiver.accept', _external=True)
+        query_params = {
+            "scope": "openid profile email",
+            # "audience": audience,
+            "response_type": "code",
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "redirect_uri": redirect_url,
+            "state": state,
+            # "prompt": "none"
+        }
+        auth_url = base_url + "?" + urlencode(query_params)
+        silent_response = requests.get(auth_url)
+
+        # return redirect('http://adr.local/user/edit/admin')
+        # return jsonify({"message": "Silent authentication initiated.", "args": request.args, "silent_response": silent_response.text})
+        return redirect(auth_url)
```

Comment:
We should use here 'Auth0 Management API'.

Please use https://auth0.com/docs/api/management/v2#!/Users/get_users_by_id endpoint. Include only following fields: `email,user_metadata`. To get `user_id` please run action `user_show` with parameter `include_plugin_extras` set to `true` for current user. This way you'll get plugin_extras, where for `saml2auth` you'll get `saml_id` which is `user_id` in Auth0.

You can copy code I used in APE to get management_token and access Management API.

Domain is same as for APE. I'll send you `client_id`, and `client_secret` via slack. They should be added as new variables to adx_config (as before).

Reply:
This is a great solution, we need to :
Configure Auth0 to send user_id as a SAML attribute
Configure adx_config.ini to map saml_id in ADR
We are running out of time so I'll ask Tomek tomorrow